docker-app/qfieldcloud/core/tests/test_jobs.py

```diff
@@ -0,0 +1,173 @@
+import logging
+
+from qfieldcloud.authentication.models import AuthToken
+from qfieldcloud.core.models import (
+    Job,
+    PackageJob,
+    Person,
+    ProcessProjectfileJob,
+    Project,
+)
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.test import APITransactionTestCase
+
+from .utils import (
+    set_subscription,
+    setup_subscription_plans,
+    testdata_path,
+    wait_for_project_ok_status,
+)
+
+logging.disable(logging.CRITICAL)
+
+VALID_LOCALIZED_LAYER_KEY = "apiary_853c29cb_ead8_4932_a60a_5cba33140b3d"
+INVALID_LAYER_KEY = "apiary_e7d1b542_1b1c_422d_ac75_c14d2b54e472"
+INVALID_LOCALIZED_LAYER_KEY = "area_a7b70aff_23f2_4876_9d4e_fa63036df0b2"
+
+
+class QfcTestCase(APITransactionTestCase):
+    def setUp(self):
+        setup_subscription_plans()
+
+        # Create a user
+        self.u1 = Person.objects.create_user(username="u1", password="abc123")
+        self.t1 = AuthToken.objects.get_or_create(user=self.u1)[0]
+
+        # Create a project
+        self.p1 = Project.objects.create(name="p1", is_public=False, owner=self.u1)
+
+    def upload_file(self, project_id: str, local: str, remote: str) -> Response:
+        """Upload a file to QFieldCloud using API.
+
+        Args:
+            project_id (str): id of the project that should contain the file.
+            local (str): name of the local file to upload, should be in `testdata` folder.
+            remote (str): name of the uploaded file.
+
+        Returns:
+            Response: response to the POST HTTP request.
+        """
+        file_path = testdata_path(local)
+        response = self.client.post(
+            f"/api/v1/files/{project_id}/{remote}/",
+            {
+                "file": open(file_path, "rb"),
+            },
+            format="multipart",
+        )
+        return response
+
+    def test_bad_layer_handler_values_for_process_projectfile_job(self):
+        # Test that BadLayerHandler is parsing data properly during process projectfile jobs
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.t1.key)
+
+        # Push the QGIS project and datasource GPKG files
+        response = self.upload_file(self.p1.id, "bumblebees.gpkg", "bumblebees.gpkg")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
```suggestion

        self.assertEqual(response.status_code, status.HTTP_201_CREATED)

```

Always add an empty line around asserts to highlight their presence. You can group multiple asserts without a newline between them.